### PR TITLE
pretty print json in debug endpoints

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
not only endpoints, all debug interface json output has no indent

**Comment by zirain** [comment]
can we make this configurable? sometime json output with indent will render too much lines, and will be difficulty to copy.

**Comment by ramaraochavali** [comment]
>not only endpoints, all debug interface json output has no indent

I meant debug endpoints not istio endpoints

**Comment by ramaraochavali** [comment]
> can we make this configurable? sometime json output with indent will render too much lines, and will be difficulty to copy.

Made it configurable via query param

File: pilot/pkg/xds/debug.go [hunk 1]
```diff
@@ -1112,9 +1114,15 @@ func (p jsonMarshalProto) MarshalJSON() ([]byte, error) {
 }
 
 // writeJSON writes a json payload, handling content type, marshaling, and errors
-func writeJSON(w http.ResponseWriter, obj interface{}) {
+func writeJSON(w http.ResponseWriter, obj interface{}, req *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
-	b, err := config.ToPrettyJSON(obj)
+	var b []byte
+	var err error
+	if req.URL.Query().Has("format") {
```
Comment (zirain) [inline pilot/pkg/xds/debug.go:1121]: rename to `pretty`?

Comment (ramaraochavali) [reply pilot/pkg/xds/debug.go:1121]: I have no strong preference. Will change

**Comment by howardjohn** [comment]
IMO this is something that is not responsibility of the server - use `jq` on client side.

Fine with it if everyone else wants this though

**Comment by ramaraochavali** [comment]
one use case I found this useful is to view smaller configs like mesh config with formatted output in the browser it self without using any other tools

**Review by hzxuzhonghu — APPROVED** [review]
LGTM

**Pull request merged**